registry/access_control/api.py

```diff
@@ -120,3 +119,13 @@ def add_userrole(project: str, user: str, role: str, reason: str, access: UserAc
 @router.delete("/users/{user}/userroles/delete", name="Delete a user role [Project Manage Access Required]")
 def delete_userrole(user: str, role: str, reason: str, access: UserAccess= Depends(project_manage_access)):
     return rbac.delete_userrole(access.project_name, user, role, reason, access.user_name)
+
+class RegistryException(HTTPException):
+    def __init__(self, status_code, detail: Any = None) -> None:
+        super().__init__(status_code=status_code,
+                         detail=detail, headers={"WWW-Authenticate": "Bearer"})
+
+def check(r):
+    if not r.ok:
+        raise RegistryException(r.status_code, f"RBAC layer failed to call registry API, error is {r.text}")
```

Comment:
maybe we can change it to :

f"Feathr registry has failures, error is {r.text}"

since we can directly expose the registry failure to end users

Reply:
Chen suggests that RBAC layer should not throw exception but respect the registry status code as well as content. The behavior now is transparent and this `RegistryException` is removed